aiohttp/client_proto.py

```diff
@@ -234,14 +271,14 @@ def data_received(self, data: bytes) -> None:
                 # parse http messages
                 try:
                     messages, upgraded, tail = self._parser.feed_data(data)
-                except BaseException as exc:
+                except BaseException as underlying_exc:
                     if self.transport is not None:
                         # connection.release() could be called BEFORE
                         # data_received(), the transport is already
                         # closed in this case
                         self.transport.close()
                     # should_close is True after the call
-                    self.set_exception(exc)
+                    self.set_exception(HttpProcessingError(), underlying_exc)
```

Comment:
@Dreamsorcerer @bdraco I made this unconditional but turned it into a different error, because this turned out the wrong level of abstraction. It seems to be turned into what makes the tests pass in the end.